torch/csrc/api/include/torch/nn/modules/dropout.h

```diff
@@ -14,34 +14,95 @@ namespace torch {
 
-/// Applies [Dropout](https://arxiv.org/abs/1207.0580) during training.
-///
-/// See https://pytorch.org/docs/stable/nn.html#torch.nn.Dropout to learn more
-/// about the exact semantics of this module.
-class TORCH_API DropoutImpl : public detail::DropoutImplBase<DropoutImpl> {
- public:
-  explicit DropoutImpl(const DropoutOptions& options_ = DropoutOptions());
+}
 
-  /// During training, applies a noise mask to the input tensor.
-  /// During evaluation, applies an identity function.
-  Tensor forward(const Tensor& input);
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Dropout ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// Applies dropout over a 1-D input.
+/// See https://pytorch.org/docs/master/nn.html#torch.nn.Dropout to learn
+/// about the exact behavior of this module.
+class TORCH_API DropoutImpl : public detail::_DropoutNd<DropoutImpl> {
+public:
+  using detail::_DropoutNd<DropoutImpl>::_DropoutNd;
+
+  Tensor forward(Tensor input);
 
   /// Pretty prints the `Dropout` module into the given `stream`.
   void pretty_print(std::ostream& stream) const override;
 };
 
+/// A `ModuleHolder` subclass for `DropoutImpl`.
+/// See the documentation for `DropoutImpl` class to learn what methods it
+/// provides, or the documentation for `ModuleHolder` to learn about PyTorch's
+/// module storage semantics.
+TORCH_MODULE(Dropout);
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Dropout2d ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// Applies dropout over a 2-D input.
+/// See https://pytorch.org/docs/master/nn.html#torch.nn.Dropout2d to learn
+/// about the exact behavior of this module.
+class TORCH_API Dropout2dImpl : public detail::_DropoutNd<Dropout2dImpl> {
+public:
+  using detail::_DropoutNd<Dropout2dImpl>::_DropoutNd;
+
+  Tensor forward(Tensor input);
+
+  /// Pretty prints the `Dropout2d` module into the given `stream`.
+  void pretty_print(std::ostream& stream) const override;
+};
+
+/// A `ModuleHolder` subclass for `Dropout2dImpl`.
+/// See the documentation for `Dropout2dImpl` class to learn what methods it
+/// provides, or the documentation for `ModuleHolder` to learn about PyTorch's
+/// module storage semantics.
+TORCH_MODULE(Dropout2d);
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Dropout3d ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// Applies dropout over a 3-D input.
+/// See https://pytorch.org/docs/master/nn.html#torch.nn.Dropout3d to learn
+/// about the exact behavior of this module.
+class TORCH_API Dropout3dImpl : public detail::_DropoutNd<Dropout3dImpl> {
+public:
+  using detail::_DropoutNd<Dropout3dImpl>::_DropoutNd;
+
+  Tensor forward(Tensor input);
+
+  /// Pretty prints the `Dropout3d` module into the given `stream`.
+  void pretty_print(std::ostream& stream) const override;
+};
+
+/// A `ModuleHolder` subclass for `Dropout3dImpl`.
+/// See the documentation for `Dropout3dImpl` class to learn what methods it
+/// provides, or the documentation for `ModuleHolder` to learn about PyTorch's
+/// module storage semantics.
+TORCH_MODULE(Dropout3d);
```

Comment:
I believe we can update the parity status of `Dropout` / `Dropout2d` / `Dropout3d` in `test/cpp_api_parity/parity-tracker.md` as well :D